### PR TITLE
QrReaderMultiBox: create list of scannedBoxes in the cache

### PR DESCRIPTION
**Pull request by HaGuesto**
done:
- create mutable shared state list in apollo cache to track scannedBoxes globally
- basic UI element for QrReaderMultiBox
- refactor of FindBoxByLabel Input as depicted in figma
- scanning a box multiple times adds the Box multiple times to the cache
- overlay closes only for single box scan

missing:
- tests

### CODE REVIEW

**Comment by pylipp** [comment]
### Functional review
I created two boxes from scratch and then tried to add them using multi-box scan from the QR scanner overlay.
The counter works as expected, also the delete button and the undo button, and the caching when closing and re-opening the overlay.
But when the second box has been scanned, I can see that the back-end still receives requests repeatedly. The camera view stays white such that I can't scan another QR code. In the single-box-scan tab the magnifying-glass-button periodally changes to a spinner. I can close the overlay and re-open it, and the requests will stop and the camera view will work again.
The same happens when I scan one box first, and then the same box again.

Btw when I use the /qrreader endpoint multi-box scan I get directed to BoxView immediately instead of staying on /qrreader in order to scan more boxes.

**Comment by HaGuesto** [comment]
> But when the second box has been scanned, I can see that the back-end still receives requests repeatedly. 

I changed it to that for the MultiBoxTab it first queries the cache.

> The camera view stays white such that I can't scan another QR code. 

I cannot reproduce that. You might want to show me that. Generally, the white screen should flare up from time to time. It indicates that the QrReaderScanner caught a Qr-Code in his video stream.
I added now a toast notification telling the user if a box that is scanned is already on the list to make it clearer when a scan is processed. Please be aware that one Qr-Code that is found by the camera is processed per second.

> In the single-box-scan tab the magnifying-glass-button periodally changes to a spinner. 

That is expected behavior. The spinner is caused by the loading state of a Qr-Code being processed.

> I can close the overlay and re-open it, and the requests will stop and the camera view will work again. The same happens when I scan one box first, and then the same box again.

You need to show me. I cannot reproduce it.

> Btw when I use the /qrreader endpoint multi-box scan I get directed to BoxView immediately instead of staying on /qrreader in order to scan more boxes.

I fixed this issue

@pylipp feel free to test again.

**Comment by pylipp** [comment]
> I added now a toast notification telling the user if a box that is scanned is already on the list to make it clearer when a scan is processed

Again, I created two boxes from scratch.
When I scan the same box twice multiple alerts pop up without stopping unless I close the overlay.

When I scan two different boxes one by one, for the second one it says "box added to list" and then again non-stop alerts "box already added to the list"

If you can't reproduce let's have a pair-session tmrw.

**Comment by HaGuesto** [comment]
[2.0 QrReaderMultiTab: create list of scanned boxes in cache.](https://trello.com/c/opqy7IWk/1160-20-qrreadermultitab-create-list-of-scanned-boxes-in-cache)

**Comment by HaGuesto** [comment]
@pylipp I resolved all the issues and I also understood the reason now. Had to dig quite deep into the QrReader library
 Please test again.

**Comment by pylipp** [comment]
> @pylipp I resolved all the issues and I also understood the reason now. Had to dig quite deep into the QrReader library Please test again.

@HaGuesto 
on Chrome, everything's fine. Single-scan & multi-scan of identical and different and non-existent boxes (alerts show up as expected and without flooding); both for /qrreader endpoint and the overlay.

on Firefox, the behavior is the same, everything's fine (though initially the QR camera field was white and I couldn't scan anything. I had to close the app in Chrome and logout and in again in the app on Firefox. Then it worked). Finally :smile:

File: react/src/components/QrReader/components/QrReaderScanner.tsx
Comment (pylipp) [inline react/src/components/QrReader/components/QrReaderScanner.tsx:42]: QrReaderScanner is also used in `PackingScanBoxOrFindByLabelOverlayContent.tsx` so you have to add the multiscan argument there.

**Comment by HaGuesto** [comment]
@pylipp I'm so happy that it works now in firefoy. That was really hard to figure out.

Yes, I would also not merge it now until we do a new production deploy

**Comment by HaGuesto** [comment]
@pylipp I fixed the error in PackingAddBoxOrItemsForPackingList. Can u mark the PR as approved?

**Review by pylipp — APPROVED** [review]

**Comment by pylipp** [comment]
@HaGuesto There's still a failing test. When it's fixed we can merge this PR and unblock e.g. #749

**Comment by HaGuesto** [comment]
@pylipp I fixed the tests. You can merge this after the v2 deploy

**Comment by aerinsol** [comment]
Good work @pylipp @HaGuesto !!

**Comment by aerinsol** [comment]
This has been merged, but I'm logging some of my concerns here as well as on the [Trello card](https://trello.com/c/opqy7IWk):

> ... none of the multi-selects (move to location, tax boxes, assign to shipment) work in terms of actual pulling of available locations and shipments.
> 
> Also, the zoom on the QR scanner doesn’t appear to be working for me either. We can disable it if it’s buggy like that.
> 
> Finally, for the multi-scan, it currently has some weird interaction if you start with multi-scan and switch to solo scan. I don’t know how to explain it except that solo scan doesn’t work properly if you try to scan something that is already cached in multiscan. Let me know if this is intended or you need more details.

**Pull request merged**
